app/controllers/users_controller.rb

```diff
@@ -46,6 +51,7 @@ def update
   # DELETE /accounts/1 or /accounts/1.json
   def destroy
     @user = User.find(params[:id])
+    PersonMilestoneMap.where(person_id: @user.id).destory_all
```

Comment:
Let me know if you want to collaborate. I can help out with moving the logic into the model.